dali/pipeline/operators/decoder/nvjpeg_decoder_decoupled_api.h

```diff
@@ -91,45 +93,64 @@ class nvJPEGDecoder : public Operator<MixedBackend>, CachedDecoderImpl {
     // GPU
     // create the handles, streams and events we'll use
     // We want to use nvJPEG default device allocator
-    for (int i = 0; i < num_threads_ * 2; ++i) {
-      NVJPEG_CALL(nvjpegJpegStreamCreate(handle_, &jpeg_streams_[i]));
-      NVJPEG_CALL(nvjpegBufferPinnedCreate(handle_, nullptr, &pinned_buffer_[i]));
-      CUDA_CALL(cudaEventCreate(&events_[i]));
+    for (auto &stream : jpeg_streams_) {
+      NVJPEG_CALL(nvjpegJpegStreamCreate(handle_, &stream));
     }
-    for (int i = 0; i < num_threads_; ++i) {
-      NVJPEG_CALL(nvjpegBufferDeviceCreate(handle_, nullptr, &device_buffer_[i]));
-      CUDA_CALL(cudaStreamCreateWithPriority(&streams_[i], cudaStreamNonBlocking,
-                                             default_cuda_stream_priority_));
+    for (auto &buffer : pinned_buffers_) {
+      NVJPEG_CALL(nvjpegBufferPinnedCreate(handle_, nullptr, &buffer));
+    }
+    for (auto &buffer : device_buffers_) {
+      NVJPEG_CALL(nvjpegBufferDeviceCreate(handle_, nullptr, &buffer));
+    }
+    for (auto &stream : streams_) {
+      CUDA_CALL(cudaStreamCreateWithFlags(&stream, cudaStreamNonBlocking));
+    }
+    for (auto &event : decode_events_) {
+      CUDA_CALL(cudaEventCreate(&event));
+      CUDA_CALL(cudaEventRecord(event, streams_[0]));
     }
-    CUDA_CALL(cudaEventCreate(&master_event_));
   }
 
   ~nvJPEGDecoder() override {
     try {
       thread_pool_.WaitForWork();
-    } catch (...) {
-      // As other images being process might fail, but we don't care
+    } catch (const std::runtime_error &) {
```

Comment:
Can we print the content of `std::runtime_error` ?

Reply:
Done.